### PR TITLE
Pass --email to bootstrap script

### LINKED ISSUE #217: Should use --email arg to bootstrap script; have user directory mounted in EFS
See https://bugzilla.mozilla.org/show_bug.cgi?id=1335441

### PR DESCRIPTION
**Pull request by jezdez**
Fix #217 
Refs https://github.com/mozilla/emr-bootstrap-spark/pull/60

### CODE REVIEW

**Comment by jezdez** [comment]
Gonna wait for @fbertsch to take a look at it first, he said in https://github.com/mozilla/emr-bootstrap-spark/pull/60#discussion_r102242995 that we need to pass in the `--efs-dns` parameter as well

File: atmo/clusters/provisioners.py [hunk 1]
```diff
@@ -39,7 +39,10 @@ def start(self, user_email, identifier, emr_release, size, public_key):
                 'Name': 'setup-telemetry-cluster',
                 'ScriptBootstrapAction': {
                     'Path': self.script_uri,
-                    'Args': ['--public-key', public_key]
+                    'Args': [
+                        '--public-key', public_key,
+                        '--email', user_email,
```
Comment (fbertsch) [inline atmo/clusters/provisioners.py]: We need to add --efs-dns as well.

Comment (fbertsch) [inline atmo/clusters/provisioners.py]: Okay, and for efs-dns, we should have separate drives for stage and prod.

**Comment by fbertsch** [comment]
@jezdez, the bootstrap script changes have been deployed.

**Comment by jezdez** [comment]
@fbertsch Cool, I'll work on updating it this week.

**Comment by fbertsch** [comment]
@jezdez do we want to update some of the text for stuff along with this change? For example, the email about the cluster being turned off doesn't need to tell people to be sure to download their files (I think it says that). And the launch page could have a quick description of this change?

**Comment by jezdez** [comment]
@fbertsch Sure, feel free to update the bug217 branch here!

**Comment by jezdez** [comment]
@fbertsch Nevermind I'm in the neighborhood and will handle the update.

File: atmo/jobs/provisioners.py [hunk 2]
```diff
@@ -56,7 +56,10 @@ def run(self, user_email, identifier, emr_release, size,
                 'Name': 'setup-telemetry-spark-job',
                 'ScriptBootstrapAction': {
                     'Path': self.script_uri,
-                    'Args': ['--timeout', str(job_timeout * 60)]
+                    'Args': [
+                        '--timeout', str(job_timeout * 60),
+                        '--email', user_email,
```
Comment (jezdez) [inline atmo/jobs/provisioners.py]: Ah, good question, should I remove it?

Comment (fbertsch) [inline atmo/jobs/provisioners.py]: Yes, @mreid-moz made a great point that people could use it to lazily rely on past state and whatnot in their bucket. We should keep all data files and outputs in scheduled jobs in s3.

**Comment by jezdez** [comment]
There is now a "What's new?" section in the footer to inform the users of changes of behavior such as the EFS mount:

screen-shot-2017-03-10-15-04-29
screen-shot-2017-03-10-15-04-53

**Comment by jezdez** [comment]
@frank r?

**Comment by jezdez** [comment]
@robhudson Can you take another look at this, the whatsnew thing is a new thing that'd be useful to check out.

File: atmo/templates/atmo/base.html [hunk 3]
```diff
@@ -55,6 +55,26 @@
...
     </div>
+
+    <div class="modal" id="whatsnew-modal" tabindex="-1" role="dialog" aria-labelledby="whatsnew-label">
+      <div class="modal-dialog" role="document">
+        <div class="modal-content">
+          <div class="modal-header">
+            <button type="button" class="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">&times;</span></button>
+            <h4 class="modal-title" id="whatsnew-label">
+              What's new?
+              <small>Notable changes and updates you should be aware off.</small>
```
Comment (robhudson) [inline atmo/templates/atmo/base.html]: s/off/of

File: atmo/templates/atmo/whatsnew/list.html [hunk 4]
```diff
@@ -0,0 +1,17 @@
+{% extends "atmo/base.html" %}
+{% load staticfiles %}
+
+{% block page_title %}What's new{% endblock %}
+
+{% block content %}
+<div class="page-header">
+  <h2>What's new <small class="hidden-xs hidden-sm">Notable changes and updates you should be aware off.</small></h2>
```
Comment (robhudson) [inline atmo/templates/atmo/whatsnew/list.html]: s/off/of

**Comment by jezdez** [comment]
@robhudson Ok, I've updated the PR to have a GitHub like alert when the data of the page has changed:

screen shot 2017-03-14 at 16 43 59-fullpage

File: atmo/clusters/provisioners.py (hunk 1 above)
Comment (jezdez) [inline atmo/clusters/provisioners.py]: @fbertsch Can you check if the values (see settings.py in this PR) are still the correct ones? Once in product we'll be able to update those values without deploys (update via Django admin).

Comment (fbertsch) [inline atmo/clusters/provisioners.py]: Yup, these are correct

**Comment by frank** [comment]
Hello, I'm sorry to say but I am not the frank you are trying to
communicate with. I claimed the 'frank' username some weeks ago as the
original account seemed to be inactive. I sincerely hope this will not
create problems for your work.

2017-03-14 16:48 GMT+01:00 Frank Bertsch <notifications@github.com>:

**Comment by jezdez** [comment]
@frank Dang, sorry for the noise!

File: atmo/decorators.py [hunk 5]
```diff
@@ -67,3 +67,29 @@ def change_permission_required(model, **params):
 
 def delete_permission_required(model, **params):
     return permission_required(full_perm(model, 'delete'), model, **params)
+
+
+def modified_date(view_func):
```
Comment (robhudson) [inline atmo/decorators.py]: This has a smell of ETag headers. I haven't read and understood everything here yet but I'm curious if we might have used ETags or `Last-Modified` header.

File: atmo/decorators.py [hunk 6]
```diff
@@ -67,3 +67,29 @@ def change_permission_required(model, **params):
...
+    X-ATMO-Modified-Date: 2017-03-14T10:48:53+00:00
+    """
+    context_var = 'modified_date'
+    header = 'X-ATMO-Modified-Date'
+
+    @wraps(view_func, assigned=available_attrs(view_func))
+    def _wrapped_view(request, *args, **kwargs):
+        response = view_func(request, *args, **kwargs)
+        # This requires the use of TemplateResponse
+        modified_date = getattr(response, 'context_data', {}).get(context_var)
```
Comment (robhudson) [inline atmo/decorators.py]: I don't see where `context_var` is coming from?

Comment (robhudson) [inline atmo/decorators.py]: D'oh, nvm. I see it up there. 😊

File: atmo/jobs/provisioners.py [hunk 7]
```diff
@@ -3,6 +3,8 @@
 # file, you can obtain one at http://mozilla.org/MPL/2.0/.
 from collections import OrderedDict
 
+import constance
```
Comment (robhudson) [inline atmo/jobs/provisioners.py]: It looks like this isn't used.

File: atmo/static/js/base.js [hunk 8]
```diff
@@ -77,10 +77,76 @@ $(function() {
...
+      return;
+    };
+    var updateModifiedDate = function() {
+      $.ajax({
+        url: window.location,
+        type: 'head',
+        success: function(res, code, xhr) {
+          var returned_modified_date = moment(xhr.getResponseHeader("X-ATMO-Modified-Date"));
+          if (!returned_modified_date.isValid()) {
+            console.log(returned_modified_date);
```
Comment (robhudson) [inline atmo/static/js/base.js]: If we leave this in maybe add some more descriptive text around it so we know what it is in the console.

File: atmo/static/js/base.js [hunk 9]
```diff
@@ -77,10 +77,76 @@ $(function() {
...
+            return;
+          }
+          var difference = returned_modified_date.diff(parsed_modified_date, 'seconds');
+          console.log(returned_modified_date, parsed_modified_date, difference);
```
Comment (robhudson) [inline atmo/static/js/base.js]: Same ^

File: atmo/decorators.py (hunk 5 above)
Comment (jezdez) [inline atmo/decorators.py]: Yeah, Etag headers are similar, except that I can't do complex comparison on them and Last-Modified headers have a weird format that I can't reliably parse on the JS side with Moment. So this invents its own header :(

**Pull request merged**